### PR TITLE
New features

### PR DESCRIPTION
**Pull request by nomisto**
- Added calculate_answers from path
- Added write_answers
- Added write_rules

### CODE REVIEW

File: src/c_clause/api/QAHandler.cpp [hunk 1]
```diff
@@ -126,6 +128,38 @@ void QAHandler::calculate_answers(std::vector<std::pair<int, int>>& queries, std
...
+	{
+		while (!util::safeGetline(file, line).eof())
+		{
+            std::vector<std::string> results = util::split(line, '\t');
+			if (results.size() == 2) {
+                // rules with stats
+                stringQueries.push_back(std::make_pair(results[0], results[1]));
+            } else {
+				std::cout << 
+                    "Unsupported Filetype, please make sure that every line either \
```
Comment (Nzteb) [inline src/c_clause/api/QAHandler.cpp:147]: The error message should be for queries not for rules.

File: src/c_clause/api/QAHandler.cpp [hunk 2]
```diff
@@ -144,6 +178,45 @@ std::vector<std::vector<std::pair<int, double>>> QAHandler::getIdxAnswers(){
 }
 
 
+void QAHandler::writeAnswers(std::string outputPath, bool strings){
```
Comment (Nzteb) [inline src/c_clause/api/QAHandler.cpp:181]: remove getIdxAnswers here.  You are using this->answers below. Also we should only use it for serializing data to python as it copies (we may have to change this)

**Review by Nzteb — COMMENTED** [review]
Thanks! Just two small things. Good that you write jsonl.

**Comment by nomisto** [comment]
Thanks! I've incorporated your comments.

**Comment by Nzteb** [comment]
Thanks!

**Pull request merged**